### ISSUE TITLE
Training Code and Pretrained Models are Available

### ISSUE THREAD

**Issue by zhanghang1989** [zhanghang1989/ResNeSt#13]
Use this issue as a pointer:

- ImageNet Training [Gluon](https://gluon-cv.mxnet.io/model_zoo/classification.html), [PyTorch](https://hangzhang.org/PyTorch-Encoding/model_zoo/imagenet.html)
- Semantic Segmentation [Gluon](https://gluon-cv.mxnet.io/model_zoo/segmentation.html), [PyTorch](https://hangzhang.org/PyTorch-Encoding/model_zoo/segmentation.html)
- Object Detection and Instance Segmentation [detectron2-ResNeSt](https://github.com/zhanghang1989/detectron2-ResNeSt)

**Comment by PistonY** [comment]
Hello, paper site is still not right and train shell/script with Gluon can not open...

**Comment by zhanghang1989** [comment]
Thanks for the feedback!

I will fix the paper link in next version. You can check out the paper at https://arxiv.org/pdf/2004.08955.pdf

GluonCV will provide the `shell` and `log` soon.

**Comment by PistonY** [comment]
Another question is the models in `Table 3` not use the same training settings right? Even the original Resnet50 use so many trciks should not only have 76.15% top1 accuracy.

And in `Table 4`, I don't know whether you use `FP16` by default, but `EfficientNet` should be in `FP32`. It's a little weird that you `#P` is several times higher than his but you're faster.

Finish reading this....I don't know how to say..plentiful tricks like `EfficientNet`.I'll try to reproduce this, I've never seen top1 acc higher than 80% on ImageNet, hoping it's not hard than `EfficientNet`.

Good luck to you and this paper.

**Comment by zhanghang1989** [comment]
EfficientNet use depth-wise convolution, which has less #P and FLOPS, but has large activations/feature-map size. This type of network is slow on GPU.

A recent paper "Designing Network Design Space" also point out that the latency on GPU has better correlation with activation than flops:

![image](https://user-images.githubusercontent.com/8041160/79878984-f7968980-83a2-11ea-9ca5-02b173ad7459.png)

**Comment by zhanghang1989** [comment]
All benchmark on Table 4 are using exactly the same setting. 

FP32, official paper implementation, single Tesla V100 GPU, batchsize=16.

**Comment by stgzr** [comment]
Does learning rate influence final accuracy? In the paper the learning rate $\eta = B/256*\eta_{base}$, $\eta_{base} = 0.1$. But in the Gloun training script the learning rate is 0.5 as batchsize=64*128=8192.

**Comment by zhanghang1989** [comment]
> Does learning rate influence final accuracy? In the paper the learning rate $\eta = B/256*\eta_{base}$, $\eta_{base} = 0.1$. But in the Gloun training script the learning rate is 0.05 * (64*128)/256 ?

To make it simple, we only need user to input **per-device** batch size and learning rate, we automatically scale up the learning rate for the user, based on the number of devices/GPUs.
https://github.com/zhanghang1989/ResNeSt/blob/master/scripts/gluon/train.py#L147

**Comment by zhanghang1989** [comment]
Resolving this issue because all models and training code has been released.
